File: zig/grains/grains.zig
const math = @import("std").math;

pub const ChessboardError = error{
    IndexOutOfBounds,
};

pub fn square(index: usize) ChessboardError!u64 {
    if (index < 1 or index > 64) {
        return ChessboardError.IndexOutOfBounds;
    }

    return math.pow(u64, 2, index - 1);
}

pub fn total() !u64 {
    var totalVal: u64 = 0;

    for (0..64) |i| {
        totalVal += try square(i + 1);
    }

    return totalVal;
}

File: go/grains/grains.go
package grains

import (
	"errors"
	"math"
)

func Square(number int) (uint64, error) {
	if number <= 0 || number > 64 {
		return uint64(0), errors.New("square number must not negative")
	}
	return uint64(math.Pow(float64(2), float64(number-1))), nil
}

func Total() uint64 {
	totalVal := uint64(0)
	for square := 1; square < 65; square++ {
		totalVal += func() uint64 {
			currVal, err := Square(square)
			if err != nil {
				panic(err)
			}

			return currVal
		}()
	}
	return totalVal
}
